standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/AbstractDatabase.java

```diff
@@ -0,0 +1,302 @@
+    if (startTime + MAX_STARTUP_WAIT < System.currentTimeMillis()) {
+      throw new RuntimeException("Container failed to be ready in " + MAX_STARTUP_WAIT/1000 +
+          " seconds");
+    }
+  }
+
+  public void cleanupDockerContainer() throws IOException, InterruptedException {
+    if (runCmdAndPrintStreams(buildRmCmd(), 600) != 0) {
+      throw new RuntimeException("Unable to remove docker container");
+    }
+  }
+
+  @Override
+  public void before() throws Exception {
+    launchDockerContainer();
+    MetastoreSchemaTool.setHomeDirForTesting();
+  }
+
+  @Override
+  public void after() {
+    if ("true".equalsIgnoreCase(System.getProperty("metastore.itest.no.stop.container"))) {
+      LOG.warn("Not stopping container " + getDockerContainerName() + " at user request, please "
+          + "be sure to shut it down before rerunning the test.");
+      return;
+    }
+    try {
+      cleanupDockerContainer();
+    } catch (InterruptedException | IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public void install() {
+    createUser();
+    installLatest();
+  }
+
+  public int createUser() {
+    return new MetastoreSchemaTool().setVerbose(verbose).run(new String[] {
+        "-createUser",
+        "-dbType", getDbType(),
+        "-userName", getDbRootUser(),
+        "-passWord", getDbRootPassword(),
+        "-hiveUser", getHiveUser(),
+        "-hivePassword", getHivePassword(),
+        "-hiveDb", getDb(),
+        "-url", getInitialJdbcUrl(),
+        "-driver", getJdbcDriver()
+    });
+  }
+
+  public int installLatest() {
+    return new MetastoreSchemaTool().setVerbose(verbose).run(new String[] {
+        "-initSchema",
+        "-dbType", getDbType(),
+        "-userName", getHiveUser(),
+        "-passWord", getHivePassword(),
+        "-url", getJdbcUrl(),
+        "-driver", getJdbcDriver(),
+        "-verbose"
+    });
+  }
+
+  public int installAVersion(String version) {
+    return new MetastoreSchemaTool().setVerbose(verbose).run(new String[] {
+        "-initSchemaTo", version,
+        "-dbType", getDbType(),
+        "-userName", getHiveUser(),
+        "-passWord", getHivePassword(),
+        "-url", getJdbcUrl(),
+        "-driver", getJdbcDriver()
+    });
+  }
+
+  public int upgradeToLatest() {
+    return new MetastoreSchemaTool().setVerbose(verbose).run(new String[] {
+        "-upgradeSchema",
+        "-dbType", getDbType(),
+        "-userName", getHiveUser(),
+        "-passWord", getHivePassword(),
+        "-url", getJdbcUrl(),
+        "-driver", getJdbcDriver()
+    });
+  }
+
+  public int validateSchema() {
+    return new MetastoreSchemaTool().setVerbose(verbose).run(new String[] {
+        "-validate",
+        "-dbType", getDbType(),
+        "-userName", getHiveUser(),
+        "-passWord", getHivePassword(),
+        "-url", getJdbcUrl(),
+        "-driver", getJdbcDriver()
+    });
+  }
```

Comment:
Possibly we could do this in separate refactoring/jira but these methods are not necessary for all consumers of this API so we could put them in a separate class/interface.